src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -97,7 +97,7 @@ public ObjectNode GetFieldRvaData(FieldDesc field)
             {
                 var pInvokeFixup = (PInvokeLazyFixupField)field;
                 PInvokeMetadata metadata = pInvokeFixup.PInvokeMetadata;
-                return NodeFactory.PInvokeMethodFixup(metadata.Module, metadata.Name);
+                return NodeFactory.PInvokeMethodFixup(metadata.Module, metadata.Name, metadata.DllImportSearchPath);
```

Comment:
Might as well change this method to take `PInvokeMetadata` directly. This list is only going to get bigger (Does the method update LastError? Do we need to do A/W probing, etc.)